Tests/UnitTests/Purchasing/OfferingsManagerTests.swift

```diff
@@ -26,6 +26,7 @@ class OfferingsManagerTests: TestCase {
     var mockOfferings: MockOfferingsAPI!
     let mockOfferingsFactory = MockOfferingsFactory()
     var mockProductsManager: MockProductsManager!
+    var mockDiagnosticsTracker: DiagnosticsTrackerType!
```

Comment:
I think we can have this be of `MockDiagnosticsTracker` type to avoid force castings below

```suggestion
    var mockDiagnosticsTracker: MockDiagnosticsTracker!
```

Reply:
I think we can't because `MockDiagnosticsTracker` is an iOS 15+ class, and this test class isn't... We could make this test class only support iOS 15+ but then we would be losing some coverage I think

Reply:
Ohh right. My mistake! 🙏